include/blazesym.h

```diff
@@ -119,143 +128,21 @@ typedef struct blaze_sym_info {
-   * loaded at 0x7fe1b2dc4000.  This base address is used to
-   * translate an address in the segment to the corresponding
-   * address in the ELF file.
-   *
-   * A loader would load an executable segment with the permission of `x`
-   * (executable).  For example, the first block is with the
-   * permission of `r-xp`.
-   */
-  uintptr_t base_address;
-} blazesym_ssc_elf;
-
-/**
- * The parameters to load symbols and debug information from a kernel.
- *
- * Use a kernel image and a snapshot of its kallsyms as a source of symbols and
- * debug information.
- */
-typedef struct blazesym_ssc_kernel {
-  /**
-   * The path of a copy of kallsyms.
-   *
-   * It can be `"/proc/kallsyms"` for the running kernel on the
-   * device.  However, you can make copies for later.  In that situation,
-   * you should give the path of a copy.
-   * Passing a `NULL`, by default, will result in `"/proc/kallsyms"`.
-   */
-  const char *kallsyms;
-  /**
-   * The path of a kernel image.
-   *
-   * The path of a kernel image should be, for instance,
-   * `"/boot/vmlinux-xxxx"`.  For a `NULL` value, it will locate the
-   * kernel image of the running kernel in `"/boot/"` or
-   * `"/usr/lib/debug/boot/"`.
-   */
-  const char *kernel_image;
-} blazesym_ssc_kernel;
-
-/**
- * The parameters to load symbols and debug information from a process.
+ * An object representing an ELF inspection source.
  *
- * Load all ELF files in a process as the sources of symbols and debug
- * information.
- */
-typedef struct blazesym_ssc_process {
-  /**
-   * It is the PID of a process to symbolize.
-   *
-   * BlazeSym will parse `/proc/<pid>/maps` and load all the object
-   * files.
-   */
-  uint32_t pid;
-} blazesym_ssc_process;
-
-/**
- * The parameters to load symbols and debug information from a gsym file.
- */
-typedef struct blazesym_ssc_gsym {
-  /**
-   * The file name of a gsym file.
-   */
-  const char *file_name;
-  /**
-   * The base address is where the file's executable segment(s) is loaded.
-   */
-  uintptr_t base_address;
-} blazesym_ssc_gsym;
-
-/**
- * Parameters of a symbol source.
+ * C ABI compatible version of [`inspect::Elf`].
  */
-typedef union blazesym_ssc_params {
-  /**
-   * The variant for [`blazesym_src_type::BLAZESYM_SRC_T_ELF`].
-   */
-  struct blazesym_ssc_elf elf;
-  /**
-   * The variant for [`blazesym_src_type::BLAZESYM_SRC_T_KERNEL`].
-   */
-  struct blazesym_ssc_kernel kernel;
+typedef struct blaze_inspect_source_elf {
   /**
-   * The variant for [`blazesym_src_type::BLAZESYM_SRC_T_PROCESS`].
-   */
-  struct blazesym_ssc_process process;
-  /**
-   * The variant for [`blazesym_src_type::BLAZESYM_SRC_T_GSYM`].
+   * The path to the binary. This member is always present.
    */
-  struct blazesym_ssc_gsym gsym;
-} blazesym_ssc_params;
-
-/**
- * Description of a source of symbols and debug information for C API.
- */
-typedef struct blazesym_sym_src_cfg {
+  char *path;
```

Comment:
should this be const? it's input param, right?

Reply:
I think so, but it's existing code and beyond of this pull request's scope.

Reply:
ok, let's just not forget to address in a follow up